modules/siddhi-extensions/event-table/src/test/java/org/wso2/siddhi/extension/eventtable/hazelcast/IndexedTableTestCase.java

```diff
@@ -0,0 +1,423 @@
+
+        executionPlanRuntime.start();
+        stockStream.send(new Object[]{"IBM", 55.6f, 200l});
+        stockStream.send(new Object[]{"IBM", 55.6f, 300l});
+        stockStream.send(new Object[]{"WSO2", 55.6f, 100l});
+        deleteStockStream.send(new Object[]{"WSO2"});
+        checkStockStream.send(new Object[]{"IBM", 100l});
+
+        SiddhiTestHelper.waitForEvents(100, 0, inEventCount, 60000);
+        Assert.assertEquals("Number of success events", 0, inEventCount.get());
+        Assert.assertEquals("Number of remove events", 0, removeEventCount);
+        Assert.assertEquals("Event arrived", false, eventArrived);
+
+        executionPlanRuntime.shutdown();
+
+    }
+
+    @Test
+    public void indexedTableTest5() throws InterruptedException {
+        log.info("indexedTableTest5");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String streams = "" +
+                "define stream StockStream (symbol string, price float, volume long); " +
+                "define stream CheckStockStream (symbol string, volume long); " +
+                "define stream DeleteStockStream (symbol string); " +
+                "@from(eventtable = 'hazelcast', instance.name = 'siddhi_instance')" +
+                "@IndexBy('symbol') " +
+                "define table StockTable (symbol string, price float, volume long); ";
+        String query = "" +
+                "@info(name = 'query1') " +
+                "from StockStream " +
+                "insert into StockTable ;" +
+                "" +
+                "@info(name = 'query2') " +
+                "from DeleteStockStream " +
+                "delete StockTable " +
+                "   on StockTable.symbol=='WSO2';" +
+                "" +
+                "@info(name = 'query3') " +
+                "from CheckStockStream join StockTable " +
+                " on CheckStockStream.symbol==StockTable.symbol " +
+                "select CheckStockStream.symbol, StockTable.volume " +
+                "insert into OutStream;";
+
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(streams + query);
+
+        executionPlanRuntime.addCallback("query3", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                if (inEvents != null) {
+                    for (Event event : inEvents) {
+                        inEventCount.incrementAndGet();
+                        switch (inEventCount.get()) {
+                            case 1:
+                                Assert.assertArrayEquals(new Object[]{"IBM", 300l}, event.getData());
+                                break;
+                            default:
+                                Assert.assertSame(1, inEventCount.get());
```

Comment:
@grainier 
Since you are already handling the case of inEventCount.get() == 1 above, I think if the code hits the default section, it will already be something other than 1. So Assert.assertSame will always result in failure. In such a case, you can use the Assert.fail() instead of doing an additional check.